feat: Add search functionality

munna-yadav opened the pull request:
This PR implements search functionality. I've added demo data for testing.
- Gives suggestion for search
- Allows to assign tags 

![image](https://github.com/user-attachments/assets/47d3abcc-be60-43d1-8b82-b9eb3f66fe2e)

![image](https://github.com/user-attachments/assets/fd2ff737-cd3e-4a9b-a0f8-2a10306411bf)
@bauchdj please review.

bauchdj commented:
bun run build error:
```sh
./src/components/animate-ui/icons/icon.tsx:16:50
Type error: '"motion/react"' has no exported member named 'AnimationControls'. Did you mean 'animationControls'?

  14 | 	type MouseEvent,
  15 | } from "react";
> 16 | import { useAnimation, type SVGMotionProps, type AnimationControls, type Variants } from "motion/react";
     |                                                  ^
  17 |
  18 | import { cn } from "@/lib/utils";
  19 |
  ```
  
  Looks like `animateControls()` function exists, but the type has changed to `LegacyAnimatonControls` as it is deprecated. Please find the non-deprecated solution.
  
I ran it in dev mode. Looks awesome! Thanks for building this. There are a couple UI state things to fix. If you start typing it says nothing is found, but is doesn't search till you press `Enter` or click it. Have it either search upon key change or have say nothing is found when no results come back from a search. I would prefer it to NOT search on every key change. Search is going to be an expensive operation across providers.

Also, the suggestions below the search input. When I clicked those, I expected it to automatically search for that. But it only changed the input value. I think we should change it to automatically submit.
When I click a suggestion box, it searches for the CURRENT text value, then replaces it with the suggestion text value. So it does search when you click it, but the state is being handled incorrect.

What do you think?

**Initial state**
<img width="895" alt="Initial State" src="https://github.com/user-attachments/assets/671a37ec-bbf7-4dc1-8b68-15f5a7987980" />

**After typing**
<img width="886" alt="image" src="https://github.com/user-attachments/assets/e1cae458-6c7f-4c15-bacd-b41ed1bc190d" />

**After clicking a suggestion**
<img width="884" alt="image" src="https://github.com/user-attachments/assets/0b275425-46aa-40ad-810d-79202d8c57a3" />

**After clicking another suggestion it searches for the previous text input???**
<img width="879" alt="image" src="https://github.com/user-attachments/assets/075f8d04-3229-4e56-821c-e5bfd5010770" />

munna-yadav commented:
https://github.com/user-attachments/assets/2810295c-d5b5-4a8a-a86d-df370e95ca2e

 fixed the issues with search such as 
- says not found even until search started.
- fixed searches fro suggestion.

and I've not modified  ```icon.tsx```  in any of my commit so don't know why it has build error.
@bauchdj

bauchdj commented:
@munna-yadav pull the last commit I made to resolve a conflict. I am reviewing this now

bauchdj reviewed (commented):
Looks good!
